sdks/python/apache_beam/pipeline.py

```diff
@@ -437,11 +438,11 @@ def visit_transform(self, transform_node):
                 output_replacements[transform_node].append((tag, replacement))
 
         if replace_input:
-          new_input = [
-              input if not input in output_map else output_map[input]
-              for input in transform_node.inputs
-          ]
-          input_replacements[transform_node] = new_input
+          new_inputs = {
+              tag: input if not input in output_map else output_map[input]
+              for (tag, input) in transform_node.main_inputs.items()
+          }
+          input_replacements[transform_node] = new_inputs
```

Comment:
I can't leave a comment on Line 274 but the type annotation for `input_replacements` there needs to change from `Dict[AppliedPTransform, Sequence[Union[pvalue.PBegin, pvalue.PCollection]]]` to `Dict[AppliedPTransform, Dict[str, Union[pvalue.PBegin, pvalue.PCollection]]]` (or `Mapping`).

Reply:
Good catch. Done.